src/platform/silabs/efr32/BLEManagerImpl.cpp

```diff
@@ -663,139 +765,207 @@ void BLEManagerImpl::HandleBootEvent(void)
 
 void BLEManagerImpl::HandleConnectEvent(volatile sl_bt_msg_t * evt)
 {
-    sl_bt_evt_connection_opened_t * conn_evt = (sl_bt_evt_connection_opened_t *) &(evt->data);
-    uint8_t connHandle                       = conn_evt->connection;
-    uint8_t bondingHandle                    = conn_evt->bonding;
 
-    ChipLogProgress(DeviceLayer, "Connect Event for handle : %d", connHandle);
+    sl_bt_evt_connection_opened_t * conn_evt = (sl_bt_evt_connection_opened_t *) &(evt->data);
 
-    AddConnection(connHandle, bondingHandle);
+    if (conn_evt->advertiser == mAdvertisingSetHandle)
+    {
+        ChipLogProgress(DeviceLayer, "Connect Event for CHIPoBLE on handle : %d", conn_evt->connection);
 
-    PlatformMgr().ScheduleWork(DriveBLEState, 0);
+        AddConnection(conn_evt->connection, conn_evt->bonding);
+        PlatformMgr().ScheduleWork(DriveBLEState, 0);
+    }
+    else
+    {
+        // We assumed we are using the side channel from the moment we
+        // are not using the CHIPoBLE service.
+        VerifyOrReturn(mBleSideChannel != nullptr);
+        ChipLogProgress(DeviceLayer, "Connect Event for SideChannel on handle : %d", conn_evt->connection);
+        mBleSideChannel->AddConnection(conn_evt->connection, conn_evt->bonding);
+        return;
+    }
 }
 
 void BLEManagerImpl::HandleConnectParams(volatile sl_bt_msg_t * evt)
 {
     sl_bt_evt_connection_parameters_t * con_param_evt = (sl_bt_evt_connection_parameters_t *) &(evt->data);
 
-    if (con_param_evt->timeout < BLE_CONFIG_TIMEOUT)
+    // If the connection belongs to the CHIPoBLE service
+
+    ChipLogProgress(DeviceLayer, "Connection Parameters Event for handle : %d", con_param_evt->connection);
+    ChipLogProgress(DeviceLayer, "Interval: %d, Latency: %d, Timeout: %d", con_param_evt->interval, con_param_evt->latency,
+                    con_param_evt->timeout);
+
+    uint16_t desiredTimeout = con_param_evt->timeout < BLE_CONFIG_TIMEOUT ? BLE_CONFIG_TIMEOUT : con_param_evt->timeout;
+
+    // For better stability, renegotiate the connection parameters if the received ones from the central are outside
+    // of our defined constraints
+    if (desiredTimeout != con_param_evt->timeout || con_param_evt->interval < BLE_CONFIG_MIN_INTERVAL ||
+        con_param_evt->interval > BLE_CONFIG_MAX_INTERVAL)
```

Comment:
I'd argue that connection intervals have not been an issue in the past. Only the timeout. This might "slow" the ble communication (and connection establishment) unnecessarily.

Reply:
This came from somewhere else, I remember it popping up in a rebase conflict and I decided to keep it but nothing more. I don't mind reverting it after I confirm where it came from.

Reply:
got it